### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by FedorSerg**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,12 +20,88 @@
...
-
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+			<version>3.2.2</version>
```
Comment (a-darashenka) [inline pom.xml]: Версию стартера не нужно указывать. Она будет браться из парент стартера, указанного в секции `<parent>`

File: pom.xml [hunk 2]
```diff
@@ -21,12 +20,88 @@
...
+		</dependency>
+
+		<!--database-->
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<version>42.7.1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.hibernate</groupId>
```
Comment (a-darashenka) [inline pom.xml]: No need to add Hibernate explicitly. It will be added by **spring-boot-starter-data-jpa** below

File: pom.xml [hunk 3]
```diff
@@ -21,12 +20,88 @@
...
+			<artifactId>hibernate-core</artifactId>
+			<version>6.4.2.Final</version>
+			<type>pom</type>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
```
Comment (a-darashenka) [inline pom.xml]: Jdbc is already included by **spring-boot-starter-data-jpa** starter

File: pom.xml [hunk 4]
```diff
@@ -21,12 +20,88 @@
...
+		<!--project lombok-->
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+			<version>1.18.30</version>
+			<scope>provided</scope>
+		</dependency>
+
+		<!--TestContainers-->
+		<dependency>
```
Comment (a-darashenka) [inline pom.xml:65]: I guess **spring-boot-starter-test** contains _mockito_ and _junit_ dependencies. Will be enough to import this starter instead

File: pom.xml [hunk 5]
```diff
@@ -21,12 +20,88 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter-engine</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.testcontainers</groupId>
+			<artifactId>junit-jupiter</artifactId>
+			<version>1.19.4</version>
```
Comment (a-darashenka) [inline pom.xml]: Common _testcontainers_ version might be added to `<properties>` section

File: src/main/java/com/example/generativeai/GenerativeaiApplication.java [hunk 6]
```diff
@@ -2,12 +2,15 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
 
 @SpringBootApplication
+@EntityScan(basePackages = {"com.example.generativeai.entity"})
```
Comment (a-darashenka) [inline src/main/java/com/example/generativeai/GenerativeaiApplication.java]: No need to add this annotation. Entities will be scanned by default with help of `@SpringBootApplication`

File: src/main/java/com/example/generativeai/entity/AuthorizedUserEntity.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Entity representing authorized user metadata.
+ */
+@Data
```
Comment (a-darashenka) [inline src/main/java/com/example/generativeai/entity/AuthorizedUserEntity.java]: Don't use `@Data` on _Entities_.
https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/

File: src/main/java/com/example/generativeai/dto/PersonDto.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.NoArgsConstructor;
+
+/**
+ * Data Transfer Object (DTO) for representing a person.
+ */
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Builder
+public class PersonDto {
```
Comment (a-darashenka) [inline src/main/java/com/example/generativeai/dto/PersonDto.java]: _Java Records_ might be used for Dtos

File: pom.xml (hunk 1 above)
Comment (FedorSerg) [reply pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (FedorSerg) [reply pom.xml]: removed the unnecessary dependency

File: pom.xml (hunk 3 above)
Comment (FedorSerg) [reply pom.xml]: removed the unnecessary dependency

File: pom.xml (hunk 4 above)
Comment (FedorSerg) [reply pom.xml:65]: removed unnecessary dependencies

File: pom.xml (hunk 5 above)
Comment (FedorSerg) [reply pom.xml]: done

File: src/main/java/com/example/generativeai/GenerativeaiApplication.java (hunk 6 above)
Comment (FedorSerg) [reply src/main/java/com/example/generativeai/GenerativeaiApplication.java]: removed

File: src/main/java/com/example/generativeai/entity/AuthorizedUserEntity.java (hunk 7 above)
Comment (FedorSerg) [reply src/main/java/com/example/generativeai/entity/AuthorizedUserEntity.java]: Thanks for the link. Fixed

File: src/main/java/com/example/generativeai/dto/PersonDto.java (hunk 8 above)
Comment (FedorSerg) [reply src/main/java/com/example/generativeai/dto/PersonDto.java]: done

**Pull request merged**